Compilation fails on linux due to compiler warning

Dear Hans-Erik,

The source code fails to build on Linux (raspbian) due to a compiler warning: 

The 'fix' for now is removing: -Werror from line 34 in: cmake/Linux.cmake
so: set(CMAKE_C_FLAGS "-Wall -Wextra -Wno-unused-parameter -Werror") 
becomes: set(CMAKE_C_FLAGS "-Wall -Wextra -Wno-unused-parameter") 

The compiler warning is caused by line 41-45 of src/osal/linux/osal_udp.c:

local = (struct sockaddr_in) {
         .sin_family = AF_INET,
         .sin_addr.s_addr = htonl(addr),
         .sin_port = htons(port),
      };


The sin struct appears to be missing a field initializer. 

Kind regards,

Sven-Erik

Thank you for your response, the full error is: 

[ 31%] Building C object src/CMakeFiles/profinet.dir/osal/linux/osal_udp.o
/home/pi/p-net/src/osal/linux/osal_udp.c: In function ‘os_udp_open’:
/home/pi/p-net/src/osal/linux/osal_udp.c:45:7: error: missing initializer for field ‘sin_zero’ of ‘struct sockaddr_in’ [-Werror=missing-field-initializers]
       };
       ^
In file included from /home/pi/p-net/src/osal/linux/osal_sys.h:27:0,
                 from /home/pi/p-net/src/osal.h:29,
                 from /home/pi/p-net/src/pf_includes.h:30,
                 from /home/pi/p-net/src/osal/linux/osal_udp.c:17:
/usr/include/netinet/in.h:246:19: note: ‘sin_zero’ declared here
     unsigned char sin_zero[sizeof (struct sockaddr) -
                   ^~~~~~~~
/home/pi/p-net/src/osal/linux/osal_udp.c: In function ‘os_udp_sendto’:
/home/pi/p-net/src/osal/linux/osal_udp.c:74:4: error: missing initializer for field ‘sin_zero’ of ‘struct sockaddr_in’ [-Werror=missing-field-initializers]
    };
    ^
In file included from /home/pi/p-net/src/osal/linux/osal_sys.h:27:0,
                 from /home/pi/p-net/src/osal.h:29,
                 from /home/pi/p-net/src/pf_includes.h:30,
                 from /home/pi/p-net/src/osal/linux/osal_udp.c:17:
/usr/include/netinet/in.h:246:19: note: ‘sin_zero’ declared here
     unsigned char sin_zero[sizeof (struct sockaddr) -
                   ^~~~~~~~
cc1: all warnings being treated as errors
src/CMakeFiles/profinet.dir/build.make:400: recipe for target 'src/CMakeFiles/profinet.dir/osal/linux/osal_udp.o' failed
make[2]: *** [src/CMakeFiles/profinet.dir/osal/linux/osal_udp.o] Error 1
CMakeFiles/Makefile2:981: recipe for target 'src/CMakeFiles/profinet.dir/all' failed
make[1]: *** [src/CMakeFiles/profinet.dir/all] Error 2
Makefile:162: recipe for target 'all' failed
make: *** [all] Error 2

--- a/src/osal/linux/osal_udp.c
+++ b/src/osal/linux/osal_udp.c
@@ -42,6 +42,7 @@ int os_udp_open(os_ipaddr_t addr, os_ipport_t port)
          .sin_family = AF_INET,
          .sin_addr.s_addr = htonl(addr),
          .sin_port = htons(port),
+         .sin_zero = { 0 },
       };
       ret = bind(id, (struct sockaddr *)&local, sizeof(local));
       if(ret != 0)
@@ -71,6 +72,7 @@ int os_udp_sendto(uint32_t id,
       .sin_family = AF_INET,
       .sin_addr.s_addr = htonl(dst_addr),
       .sin_port = htons(dst_port),
+      .sin_zero = { 0 },
    };
    len = sendto(id, data, size, 0, (struct sockaddr *)&remote, sizeof(remote));
 

